sale_tier_validation/models/res_config_settings.py

```diff
@@ -0,0 +1,34 @@
+# Copyright 2024 Quartile (https://www.quartile.co)
+# License AGPL-3.0 or later (https://www.gnu.org/licenses/agpl.html).
+
+from odoo import api, fields, models
+
+
+class ResConfigSettings(models.TransientModel):
+    _inherit = "res.config.settings"
+
+    sale_report_print_block = fields.Boolean(
+        help="Block the printing of the sale order report if the order is not validated."
+    )
```

Comment:
It makes more sense to me to make it configurable at company level, and have it disabled by default to avoid the chances of interfering with tests.
```suggestion
    sale_report_print_block = fields.Boolean(
        related="company_id.sale_report_print_block",
        help="Block the printing of the sale order report if the order is not validated."
    )
```